### ISSUE TITLE
Only one match

### ISSUE THREAD

**Issue by bfialkoff** [nmslib/hnswlib#394]
I am trying to use this for keypoint feature matching so I know that each query can only be matched to one entry in the Index, is there any way to enforce this behavior?

**Comment by yurymalkov** [comment]
Hi @bfialkoff, I think so. That sound like k=1

**Comment by bfialkoff** [comment]
I don't think so, k=1 means that itll give me only the top nearest vector in the Index. 
I'm trying to query with N vectors, such that if one of my vectors has been matched to vector `i` in the index, no other query vector will be mapped to vector `i` in the index.

**Comment by yurymalkov** [comment]
Oh, I see. I do not think there is currently such a functionality without serious C++ code modification as queries are now processed independently from each other.
